dart/collision/CollisionGroup.h

```diff
@@ -0,0 +1,210 @@
+  void registerShapeFramesOf(const dynamics::ShapeFrame* shapeFrame);
+
+  /// As as registerShapeFrames()
+  void registerShapeFramesOf(
+      const std::vector<const dynamics::ShapeFrame*>& shapeFrames);
+
+  /// Register all the ShapeFrames in the other CollisionGroup to this
+  /// CollisionGroup
+  void registerShapeFramesOf(const CollisionGroup* other);
+
+  /// Register all the BodyNode's ShapeFrames having CollisionAddon to this
+  /// CollisionGroup
+  void registerShapeFramesOf(const dynamics::BodyNode* bodyNode);
+
+  /// Register all the Skeleton's ShapeFrames having CollisionAddon to this
+  /// CollisionGroup
+  void registerShapeFramesOf(const dynamics::Skeleton* skeleton);
+
+  /// Unregister a ShapeFrame from this CollisionGroup
+  void unregisterShapeFrame(const dynamics::ShapeFrame* shapeFrame);
+
+  /// Unregister ShapeFrames from this CollisionGroup
+  void unregisterShapeFrames(
+      const std::vector<const dynamics::ShapeFrame*>& shapeFrames);
+
+  /// Unregister all the ShapeFrames in the other CollisionGroups from this
+  /// CollisionGroup
+  template <typename First, typename... Others>
+  void unregisterShapeFrames(const First* first, const Others*... others);
+
+  /// Do nothing. This function is for terminating the variadic template
+  /// function template<typename...> unregisterShapeFrames().
+  void unregisterShapeFramesOf();
+
+  /// As as unregisterShapeFrame()
+  void unregisterShapeFramesOf(const dynamics::ShapeFrame* shapeFrame);
+
+  /// As as unregisterShapeFrames()
+  void unregisterShapeFramesOf(
+      const std::vector<const dynamics::ShapeFrame*>& shapeFrames);
+
+  /// Unregister all the ShapeFrames in the other CollisionGroup from this
+  /// CollisionGroup
+  void unregisterShapeFramesOf(const CollisionGroup* other);
+
+  /// Unregister all the BodyNode's ShapeFrames having CollisionAddon from this
+  /// CollisionGroup
+  void unregisterShapeFramesOf(const dynamics::BodyNode* bodyNode);
+
+  /// Unregister all the Skeleton's ShapeFrames having CollisionAddon from this
+  /// CollisionGroup
+  void unregisterShapeFramesOf(const dynamics::Skeleton* skeleton);
+
+  /// Unregister all the ShapeFrames in this CollisionGroup
+  void unregisterAllShapeFrames();
+
+  /// Return true if this CollisionGroup contains shapeFrame
+  bool hasShapeFrame(const dynamics::ShapeFrame* shapeFrame) const;
+
+  /// Return number of ShapeFrames added to this CollisionGroup
+  size_t getNumShapeFrames() const;
+
+  /// Return index-th ShapeFrames registered to this CollisionGroup. Return
+  /// nullptr if the index is out of the range.
+  const dynamics::ShapeFrame* getShapeFrame(size_t index) const;
+
+  /// Return all the ShapeFrames registered to this CollisionGroup
+  const std::vector<const dynamics::ShapeFrame*> getShapeFrames() const;
+
+  /// Update engine data. This function should be called before the collision
+  /// detection is performed by the engine in most cases.
+  void update();
```

Comment:
What's the motivation behind making this `public`, instead of `protected` like the other internal methods used by the `CollisionDetector`? Is there any case where I would call this as a user?